### PR TITLE
[TTAHUB-5229] Update fact tables last TTA source

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

In the UI/previous iterations of the monitoring fact tables, "last tta date" represented the latest end date of a report attached to a citation. The fact tables seem to use start date.

## How to test

Confirm changes correctly source "last_tta" from the proper column. Data updates will occur as the monitoring script runs.

## Issue(s)

- https://jira.acf.gov/browse/TTAHUB-5229

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete
- [ ] QA review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] PR created as **Draft**
- [ ] Staging smoke test completed
- [ ] PR transitioned to **Open**
- [ ] Reviewer added _(after transitioning to Open to ensure Slack notifications trigger; `elainaparrish` is the authorized approver under normal circumstances)_
  - _Sequence: Draft PR → Smoke test → Open PR → Add reviewer_
  - _Confirm that Slack notification was sent after reviewer was added_

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by Andrew565 — APPROVED** [review]
Approved but probably would be good to add copilot's suggested tests.

File: src/tools/updateMonitoringFactTables.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ export const recreateLiveValuesViews = async (
     WITH last_ar AS (
     SELECT DISTINCT ON (drc."deliveredReviewId")
       drc."deliveredReviewId" ar_drid,
-      ar."startDate" last_tta,
+      ar."endDate" last_tta,
```
Comment (kryswisnaskas) [inline src/tools/updateMonitoringFactTables.ts:82]: Codex said:

there is still no regression test for the SQL views that this PR changes. The existing tests run updateMonitoringFactTables, but they never assert Citation.scope('withLiveValues') or DeliveredReview.scope('withLiveValues') for last_tta / last_ar_id; the end-date coverage that exists today is widget-level, not view-level. That means this exact bug can come back without any failing backend test. Add a focused test with two approved ARs on the same citation/review, different start/end dates, plus at least one null or tie case, and assert both live-value views return the expected last_tta and matching last_ar_id.

File: src/tools/updateMonitoringFactTables.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const recreateLiveValuesViews = async (
     WITH last_ar AS (
     SELECT DISTINCT ON (aroc."citationId")
       aroc."citationId" ar_cid,
-      ar."startDate" last_tta,
+      ar."endDate" last_tta,
```
Comment (kryswisnaskas) [inline src/tools/updateMonitoringFactTables.ts:32]: Codex said:

this PR changes the fact-table definition of “Last TTA” to end-date semantics, but recipientSpotlight still computes last_tta from MAX(ar."startDate"). That leaves the application with two different definitions of the same field after merge, so users can still see conflicting “Last TTA” values across screens. The rest of the monitoring code already treats last TTA as an end date (src/services/monitoring.ts (line 401)). Fix by updating recipientSpotlight to the same end-date rule, or document very clearly why that endpoint is intentionally different.

Comment (thewatermethod) [reply src/tools/updateMonitoringFactTables.ts:32]: recipient spotlight does not use the fact tables and is unrelated functionally. I'd be glad to tackle that if the work is prioritized in a ticket; I'm making the fix to the fact tables since they are directly related to the monitoring dashboard sorting

Comment (thewatermethod) [reply src/tools/updateMonitoringFactTables.ts:32]: logged this as a bug for future work: https://jira.acf.gov/browse/TTAHUB-5313

File: src/tools/updateMonitoringFactTables.ts (hunk 1 above)
Comment (thewatermethod) [reply src/tools/updateMonitoringFactTables.ts:82]: Added tests [a1d9ba9](https://github.com/HHS/Head-Start-TTADP/pull/3601/commits/a1d9ba98cb8b3e84d1cc2824088294b6ed3d8d03) & [701a0c3](https://github.com/HHS/Head-Start-TTADP/pull/3601/commits/701a0c308111c459c4e2908c211856ca912a2c0a)

File: src/tools/updateMonitoringFactTables.ts [hunk 3]
```diff
@@ -79,7 +79,7 @@ export const recreateLiveValuesViews = async (
     WITH last_ar AS (
     SELECT DISTINCT ON (drc."deliveredReviewId")
       drc."deliveredReviewId" ar_drid,
-      ar."startDate" last_tta,
+      ar."endDate" last_tta,
       ar.id last_ar_id
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply src/tools/updateMonitoringFactTables.ts:83]: Added tests [a1d9ba9](https://github.com/HHS/Head-Start-TTADP/pull/3601/commits/a1d9ba98cb8b3e84d1cc2824088294b6ed3d8d03) & [701a0c3](https://github.com/HHS/Head-Start-TTADP/pull/3601/commits/701a0c308111c459c4e2908c211856ca912a2c0a)

File: src/tools/updateMonitoringFactTables.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export const recreateLiveValuesViews = async (
     WITH last_ar AS (
     SELECT DISTINCT ON (aroc."citationId")
       aroc."citationId" ar_cid,
-      ar."startDate" last_tta,
+      ar."endDate" last_tta,
       ar.id last_ar_id
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply src/tools/updateMonitoringFactTables.ts:33]: Added tests [a1d9ba9](https://github.com/HHS/Head-Start-TTADP/pull/3601/commits/a1d9ba98cb8b3e84d1cc2824088294b6ed3d8d03) & [701a0c3](https://github.com/HHS/Head-Start-TTADP/pull/3601/commits/701a0c308111c459c4e2908c211856ca912a2c0a)

**Review by AdamAdHocTeam — APPROVED** [review]

File: src/tools/updateMonitoringFactTables.ts (hunk 2 above)
Comment (hardwarehuman) [reply src/tools/updateMonitoringFactTables.ts:32]: I don't have any particular beliefs on whether `startDate` or `endDate` is correct, but I seem to remember meaning to match recipientSpotlight to the last_tta definition from the SSDI no-tta.sql query. However, I just re-checked, and no-tta.sql uses `endDate`. Thus, I think the recipientSpotlight using `startDate` is just a straightforward bug that then replicated into the fact tables because when I was writing the view I checked the recipientSpotlight definitions.

TL;DR: `endDate` looks correct, and given that I'm refactoring recipientSpotlight to use the fact tables now, it will match imminently.

**Review by hardwarehuman — APPROVED** [review]
I was going to say the tests were awfully heavy, but I see that Copilot asked for them this way, so 🤷
